### DISCUSSION TITLE
Develop API - artist['songcount'] any issues?

### DISCUSSION THREAD

**Discussion by Andrew-McGee** [ampache/ampache#2862] [Q&A]
I have some artists where the song_count number stored in the db seems to be wrong (too low). 

For example one artist I know definitely has two songs in the library. If I call artist_albums I get 2 album results returned for them. But when I call artist and check songcount it is set to 1. I checked the db in the artist table and confirm it is stored as 1 in the song_count field.

Is there  a process that runs to update this? Perhaps it's a recently added album. How does it normally get updated?

**Comment by usox** [comment]
From what I can see, the song count of artists gets updated when:
- the artists current song_count is 0 _or_
- the artists playing time is 0_or_
- the last update of the artist is older than 1 month_or_
- `Add`-catalog-process runs _or_
- Update-From-Tags process is invoked

**Comment by Andrew-McGee** [comment]
Thanks usox. I think there might be a problem. I'm adding albums using 'add' in the GUI and it doesn't seem to be incrementing.

**Comment by Andrew-McGee** [comment]
BTW, I use this number in my frontend to display all tracks by an artist. The number is used in a loop to display table rows so if it's short some tracks get missed. I think I will change the loop to use 'for each' number of entries returned by artist_songs. That way it's always set to the actual songs returned by the API call.

**Reply by lachlan-00** [reply]
@usox is correct for all of those.
a catalog add will update all the numbers for artists with new tracks or stale (> 1 month) artists
whenever a clean is completed it will update the numbers as well. (e.g you delete tracks from this artist so redo their counts.)
The numbers are meant to be dynamic (when 0) on load as well but if they aren't for you it would be good for me to test out why.

If it's from api use only that might be something for me to look at

**Reply by Andrew-McGee** [reply]
Doesn't seem to be updating for me. I just made sure I was on the latest develop. Added an album by running update in the web GUI (it was a various artists compilation in case that changes anything) and checked a particular artist on the album but the song_count didn't increment. If I query that artist albums I get 2 coming back (compilation or otherwise) so I would also expect the song_count to be 2 but it's still set to 1. Something is up with it.

**Reply by lachlan-00** [reply]
Checked after add using the API or in web UI?

Keep these checks coming!

**Reply by Andrew-McGee** [reply]
Added in the web UI. Noticed in my code but then checked in the db. Here are some screenshots to ensure I'm not going crazy:

Song table showing artist id = 2200 you can see 2 titles
![Overview](https://i.postimg.cc/g0hjGHZB/song-table.png)

Now the artist table for id 2200 only shows song_count of 1.
![Overview](https://i.postimg.cc/wjCx4RG5/artist-table.png)
